### DISCUSSION TITLE
CURL fails on workflow but succeeds locally

### DISCUSSION THREAD

**Discussion by mauricioklein** [community/community#27202] [Actions]
<p>Hello!</p>
<p>I’m trying to create a workflow that will trigger a Github Pages build daily for my personal blog.</p>
<p>I have two triggers for the workflow:</p>
<ul>
<li>One scheduled (once a day)</li>
<li>One manual, triggered directly in Actions UI</li>
</ul>
<p>For this, I have created a PAT (called <code>PAGES_TOKEN</code>) with all <code>repo</code> permissions.</p>
<p>My workflow file is the following:</p>
<pre><code class="lang-yaml">name: Build

on:
  workflow_dispatch:
  schedule:
    - cron: '0 15 * * *' # Runs every day at 3pm GMT (4pm CEST)

jobs:
  build-pages:
    runs-on: ubuntu-20.04
    steps:
      - name: Trigger GitHub pages rebuild
        run: |
          curl --fail --request POST --header "Authorization: Bearer $USER_TOKEN" --url "https://api.github.com/repos/${{ github.repository }}/pages/builds"
        env:
          USER_TOKEN: ${{ secrets.PAGES_TOKEN }}
</code></pre>
<p>Every time I trigger a manual build in actions UI against the <code>master</code> branch I get a 401 error back:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/7/c725888c52ead2b33fb205ad93e528c1771d0357.png" data-download-href="/uploads/short-url/spJqKPQqYscFvHKCqtc6fIyfTn1.png?dl=1" title="Screen Shot 2020-12-08 at 10.30.25" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/11538662/181102226-00473753-e8e3-43a7-bb21-379152eb7ff2.png" alt="Screen Shot 2020-12-08 at 10.30.25" data-base62-sha1="spJqKPQqYscFvHKCqtc6fIyfTn1" width="690" height="129" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c725888c52ead2b33fb205ad93e528c1771d0357_2_690x129.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c725888c52ead2b33fb205ad93e528c1771d0357_2_1035x193.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/7/c725888c52ead2b33fb205ad93e528c1771d0357.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c725888c52ead2b33fb205ad93e528c1771d0357_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-12-08 at 10.30.25</span><span class="informations">1162×218 25.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>However, if I run the exact same CURL command locally, replacing the <code>$USER_TOKEN</code> by the actual token secret, the requests succeeds with a status <code>QUEUED</code>.</p>
<p>Am I missing something in my workflow setup?</p>
<p>Thanks in advance for any hint!</p>

**Comment by airtower-luna** [comment]
<p>The <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps">documentation on token scopes</a> says to use tokens with <code>token</code>, not <code>Bearer</code>. That has worked for me so far:</p>
<pre><code class="lang-auto">curl --header "Authorization: token $USER_TOKEN" # ...
</code></pre>

**Comment by mauricioklein** [comment, marked as answer]
<p>Thanks, but Bearer works just fine as well :slightly_smiling_face:</p>
<p>Turns out the problem was a misunderstanding from my side:</p>
<p>While PATs are created in the GH account context, they need to be added as a secret to the repository to be used on workflows.</p>
<p>Adding a secret to the repo with the PAT secret solved the problem.</p>
<p>Thanks!</p>
